testing/tests/DevExpress.ui.widgets.form/formLayoutManager.markup.tests.js

```diff
@@ -3027,3 +2991,92 @@ QUnit.module('Supported editors', () => {
+                editorType: supportedEditor,
+                editorOptions: { readOnly: false }
+            }))
+        });
+
+        checkSupportedEditors((editor, className) => {
+            assert.notOk($testContainer.find(`.${FIELD_ITEM_CLASS} .${className}`).hasClass(READONLY_STATE_CLASS), `${editor}: editor is not read only`);
+        });
+    });
+
+    test('editors should be read only when readOnly option is enabled in widget and in the editorOptions is not set', function(assert) {
+        const $testContainer = $('#container').dxLayoutManager({
+            readOnly: true,
+            items: supportedEditors.map(supportedEditor => ({
+                name: supportedEditor,
+                editorType: supportedEditor,
+            }))
+        });
+
+        checkSupportedEditors((editor, className) => {
+            assert.ok($testContainer.find(`.${FIELD_ITEM_CLASS} .${className}`).hasClass(READONLY_STATE_CLASS), `${editor}: editor is read only`);
+        });
+    });
+
+    test('editors should change their readonly state after change readOnly option in widget', function(assert) {
+        const $testContainer = $('#container').dxLayoutManager({
+            readOnly: true,
+            items: supportedEditors.map(supportedEditor => ({
+                dataField: supportedEditor,
+                editorType: supportedEditor,
+            }))
+        });
+
+        checkSupportedEditors((editor, className) => {
+            assert.ok($testContainer.find(`.${FIELD_ITEM_CLASS} .${className}`).hasClass(READONLY_STATE_CLASS), `${editor}: editor is read only`);
+        });
+
+        $testContainer.dxLayoutManager('instance').option('readOnly', false);
+
+        checkSupportedEditors((editor, className) => {
+            assert.notOk($testContainer.find(`.${FIELD_ITEM_CLASS} .${className}`).hasClass(READONLY_STATE_CLASS), `${editor}: editor is not read only`);
+        });
+    });
+
+    test('editors should not change their readonly state after change readOnly option in widget if readOnly option is also set in editors options', function(assert) {
+        const $testContainer = $('#container').dxLayoutManager({
+            readOnly: false,
+            items: supportedEditors.map(supportedEditor => ({
+                dataField: supportedEditor,
+                editorType: supportedEditor,
+                editorOptions: { readOnly: false }
+            }))
+        });
+
+        checkSupportedEditors((editor, className) => {
+            assert.notOk($testContainer.find(`.${FIELD_ITEM_CLASS} .${className}`).hasClass(READONLY_STATE_CLASS), `${editor}: editor is not read only`);
+        });
+
+        $testContainer.dxLayoutManager('instance').option('readOnly', true);
+
+        checkSupportedEditors((editor, className) => {
+            assert.notOk($testContainer.find(`.${FIELD_ITEM_CLASS} .${className}`).hasClass(READONLY_STATE_CLASS), `${editor}: editor is not read only`);
+        });
+    });
+});
```

Comment:
Does it work ok for a `disabled` prop?

Reply:
The `disabled` state work different way. Disabled form has class on the whole widget and not pass disabled state to the editor. So you can change editor's disabled state apart of the form.